apps/sports-helsinki/src/domain/search/combinedSearch/adapters/CombinedSearchFormAdapter.ts

```diff
@@ -93,35 +94,56 @@ class CombinedSearchFormAdapter
     this.language = locale;
     this.geolocation = geolocation;
 
-    this.text = input.get('text') ?? input.get('q') ?? '';
-    this.venueOrderBy =
-      input.get('venueOrderBy') ??
-      input.get('orderBy') ??
-      input.get('sort') ??
-      initialCombinedSearchFormValues.venueOrderBy;
-    this.eventOrderBy =
-      input.get('eventOrderBy') ??
-      input.get('orderBy') ??
-      input.get('sort') ??
-      initialCombinedSearchFormValues.eventOrderBy;
-    this.courseOrderBy =
-      input.get('courseOrderBy') ??
-      input.get('orderBy') ??
-      input.get('sort') ??
-      initialCombinedSearchFormValues.courseOrderBy;
-    this.sportsCategories = input.getAll('sportsCategories');
-    this.targetGroups = input.getAll('targetGroups');
-    this.helsinkiOnly =
-      input.get('helsinkiOnly') ?? initialCombinedSearchFormValues.helsinkiOnly;
-    this.organization =
-      input.get('organization') ??
-      input.get('publisher') ??
-      initialCombinedSearchFormValues.organization;
-    this.place =
-      input.get('place') ??
-      input.get('places') ??
-      initialCombinedSearchFormValues.place;
-    this.keywords = input.getAll('keywords');
+    // Get a single value (i.e. not an array) type URL search parameter with fallbacks
+    function getOne<T extends CombinedSearchAdapterInputField>(
+      primaryField: T,
+      ...fallbackFields: CombinedSearchAdapterInputFallback[]
+    ) {
+      for (const field of [primaryField, ...fallbackFields]) {
+        const value = input.get(field);
+        if (value !== null) {
+          return value;
+        }
+      }
+      return initialCombinedSearchFormValues[primaryField];
+    }
+
+    // Get an array type URL search parameter with fallbacks
+    function getArray<T extends CombinedSearchAdapterInputField>(
+      primaryField: T,
+      ...fallbackFields: CombinedSearchAdapterInputFallback[]
+    ) {
+      for (const field of [primaryField, ...fallbackFields]) {
+        const value = input.getAll(field);
+        if (value.length > 0) {
+          return value;
+        }
+      }
+      return initialCombinedSearchFormValues[primaryField];
+    }
```

Comment:
These should not be inside the constructor, because they make the constructor really really hard to read and understand. They could be moved to some utils-file or made private functions of the class - could even be prefixed with an underscore ("_").

Reply:
Taken into account in 6996d2d1aa1b8314db0712955aac81434352a17d